### ISSUE TITLE
uninitialized constant SyntaxTree::Pattern::PP When Running stree search

### ISSUE THREAD

**Issue by gwillcox-r7** [ruby-syntax-tree/syntax_tree#344]
Currently I'm trying to process the following file:

```
# Sourced from Ruby's ext/pty/lib/expect.rb to allow for access from Windows,
# which does not seem to have an issue using this particular method with
# sockets (pipes and other handles won't work, so don't use it for that).
# frozen_string_literal: true
$expect_verbose = false

# Expect library adds the IO instance method #expect, which does similar act to
# tcl's expect extension.
#
# In order to use this method, you must require expect:
#
#   require 'expect'
#
# Please see #expect for usage.
class IO
  # call-seq:
  #   IO#expect(pattern,timeout=9999999)                  ->  Array
  #   IO#expect(pattern,timeout=9999999) { |result| ... } ->  nil
  #
  # Reads from the IO until the given +pattern+ matches or the +timeout+ is over.
  #
  # It returns an array with the read buffer, followed by the matches.
  # If a block is given, the result is yielded to the block and returns nil.
  #
  # When called without a block, it waits until the input that matches the
  # given +pattern+ is obtained from the IO or the time specified as the
  # timeout passes. An array is returned when the pattern is obtained from the
  # IO. The first element of the array is the entire string obtained from the
  # IO until the pattern matches, followed by elements indicating which the
  # pattern which matched to the anchor in the regular expression.
  #
  # The optional timeout parameter defines, in seconds, the total time to wait
  # for the pattern.  If the timeout expires or eof is found, nil is returned
  # or yielded.  However, the buffer in a timeout session is kept for the next
  # expect call.  The default timeout is 9999999 seconds.
  def expect(pat,timeout=9999999)
    buf = ''.dup
    case pat
    when String
      e_pat = Regexp.new(Regexp.quote(pat))
    when Regexp
      e_pat = pat
    else
      raise TypeError, "unsupported pattern class: #{pat.class}"
    end
    @unusedBuf ||= ''
    while true
      if not @unusedBuf.empty?
        c = @unusedBuf.slice!(0)
      elsif !IO.select([self],nil,nil,timeout) or eof? then
        result = nil
        @unusedBuf = buf
        break
      else
        c = getc
      end
      buf << c
      if $expect_verbose
        STDOUT.print c
        STDOUT.flush
      end
      if mat=e_pat.match(buf) then
        result = [buf,*mat.captures]
        break
      end
    end
    if block_given? then
      yield result
    else
      return result
    end
    nil
  end
end
```

I try to search for `result` but this results in an uninitalized constant error and associated stack trace:

```
> stree search result lib/expect.rb
/home/gwillcox/.rbenv/versions/3.0.5/gemsets/metasploit-framework/gems/syntax_tree-6.0.2/lib/syntax_tree/pattern.rb:92:in `compile_error': uninitialized constant SyntaxTree::Pattern::PP (NameError)
	from /home/gwillcox/.rbenv/versions/3.0.5/gemsets/metasploit-framework/gems/syntax_tree-6.0.2/lib/syntax_tree/pattern.rb:283:in `compile_node'
	from /home/gwillcox/.rbenv/versions/3.0.5/gemsets/metasploit-framework/gems/syntax_tree-6.0.2/lib/syntax_tree/pattern.rb:73:in `compile'
	from /home/gwillcox/.rbenv/versions/3.0.5/gemsets/metasploit-framework/gems/syntax_tree-6.0.2/lib/syntax_tree/cli.rb:264:in `initialize'
	from /home/gwillcox/.rbenv/versions/3.0.5/gemsets/metasploit-framework/gems/syntax_tree-6.0.2/lib/syntax_tree/cli.rb:510:in `new'
	from /home/gwillcox/.rbenv/versions/3.0.5/gemsets/metasploit-framework/gems/syntax_tree-6.0.2/lib/syntax_tree/cli.rb:510:in `run'
	from /home/gwillcox/.rbenv/versions/3.0.5/gemsets/metasploit-framework/gems/syntax_tree-6.0.2/exe/stree:9:in `<top (required)>'
	from /home/gwillcox/.rbenv/versions/3.0.5/gemsets/metasploit-framework/bin/stree:25:in `load'
	from /home/gwillcox/.rbenv/versions/3.0.5/gemsets/metasploit-framework/bin/stree:25:in `<main>'
```

**Comment by gwillcox-r7** [comment]
Woops I realized that I may have misunderstood the intent of some of the parameters so if this is the case please feel free to correct me.

**Comment by kddnewton** [comment]
Hey @gwillcox-r7! So a couple of things are going on. First, you got hit by a bug that I just fixed in 6.1.0 that had a missing require to `pp`. So that's what you're seeing. What you should have seen was:

```
Syntax Tree was unable to compile the pattern you provided to search
into a usable expression. It failed on to understand the node
represented by:

(var_field (ident "result"))

Note that not all syntax supported by Ruby's pattern matching syntax
is also supported by Syntax Tree's code search. If you're using some
syntax that you believe should be supported, please open an issue on
GitHub at https://github.com/ruby-syntax-tree/syntax_tree/issues/new.
```

The argument that gets passed to `stree search` should be a pattern that you're looking for. By putting in `result`, Syntax Tree thinks you mean to look for something equivalent to:

```ruby
case node
in result
end
```

which would effectively match any node. I'm assuming in this case you want to find references to the local `result` variable, in which case you could search for:

```
$ exe/stree search 'VarRef[value: Ident[value: "result"]]' lib/expect.rb
lib/expect.rb:68:12:       yield result
lib/expect.rb:70:13:       return result
```

or maybe you're searching for definitions, as in:

```
$ exe/stree search 'VarField[value: Ident[value: "result"]]' lib/expect.rb
lib/expect.rb:63:8:         result = [buf,*mat.captures]
lib/expect.rb:51:8:         result = nil
```

or both together, as in:

```
$ exe/stree search 'VarRef[value: Ident[value: "result"]] | VarField[value: Ident[value: "result"]]' lib/expect.rb
lib/expect.rb:68:12:       yield result
lib/expect.rb:63:8:         result = [buf,*mat.captures]
lib/expect.rb:70:13:       return result
lib/expect.rb:51:8:         result = nil
```

Hopefully that clarifies things. Let me know if there's anything else I can clarify!
